### DISCUSSION TITLE
Deploy Times

### DISCUSSION THREAD

**Discussion by jlaura** [developmentseed/titiler#476] [Q&A] [answered]
I am working through a deploy to AWS Lambda using the AWS-CDK. What kind of runtimes should I be expecting for the `npm run cdk bootstrap` command? Currently this has been running at 100% for over 60 minutes. I realize this is a one time thing, so I'm worried if it takes quite a while. I do though, want to make sure that something is to hanging in the background. Thanks!

**Comment by vincentsarago** [comment, marked as answer]
@jlaura it shouldn't take more than 5min 🤷 

I've just updated the docker image and the package-lock.json but not sure if it will change something.

can you try building the docker 
```
cd deployment/aws
docker build --platform=linux/amd64 -f lambda/Dockerfile . --no-cache
```

☝️ should be the longest step of the deployment

**Reply by jlaura** [reply]
Success. I had to modify the docker command to be:

`RUN pip install titiler.application==0.4.2 "mangum>=0.10.0" idna==3.3 importlib_resources==5.7.0 fastapi==0.77.1 click==8.1.3 snuggs==1.4.7 rasterio==1.2.9 numpy==1.22.3 -t /asset --no-binary pydantic`

So, lots more specific dependencies. This was to combat pip failures. I wonder if others don't have cached versions of code that is allowing pip to solve more smoothly?

**Reply by vincentsarago** [reply]
@jlaura really sorry about that. Next time please mention the titiler version so I can assist better 😄

**Reply by jlaura** [reply]
will do! and no worries - I really appreciate the project and am happy if the info helps anyone else.
